src/controller/errorController.js

```diff
@@ -5,13 +5,16 @@ const {
 const { handleValidationError } = require('../helpers/handleValidationError');
 
 const errorController = (err, req, res, next) => {
-  if (err.status === 401) {
-    res.status(err.status).send({
-      messages: err.message,
-    });
-  }
-
   try {
+    if (err.status === 401) {
+      res.status(err.status).send({
+        messages: err.message,
+      });
+    }
+    if (err?.kind === 'ObjectId') {
+      res.status(422).send({ messages: err.reason.message });
+    }
+
```

Comment:
What does err.kind do?

Reply:
You can check the type or error:
1. https://mongoosejs.com/docs/api.html#error_Error-ValidatorError
2. https://mongoosejs.com/docs/validation.html#validation-errors

Reply:
you are an error profi now :)